version: "3.8"

services:
  # app:
  #   build: .
  #   container_name: transactions_app
  #   volumes:
  #     - .:/go/src/
  #   ports:
  #     - 3150:8000
  #   depends_on:
  #     - db
  #   extra_hosts:
  #     - "host.docker.internal:172.17.0.1"
  #   # networks:
  #   #   - cred_system_network

  db:
    build: .docker/postgres
    container_name: transact_db
    restart: always
    tty: true
    ports:
      - 3151:5432
    volumes:
      - .docker/postgres/dbdata:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=transactions
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=thisislocal-noneedtoworry
    extra_hosts:
      - "host.docker.internal:172.17.0.1"
    # networks:
    #   - cred_system_network

  # db-oracle:
  #   build: .docker/oracle
  #   container_name: transact_oracle_db
  #   environment:
  #     - ORACLE_SID=ORCLCDB
  #     - ORACLE_PDB=ORCLPDB1
  #     - ORACLE_PWD=ThisIsLocal101NoNeed2Worry
  #   ports:
  #     - 3152:1521
  #   volumes:
  #     - oracle-data:/opt/oracle/oradata
  #     - oracle-backup:/opt/oracle/backup
  #   healthcheck:
  #     test: ["CMD", "sqlplus", "-L", "sys/ThisIsLocal101NoNeed2Worry@//localhost:1521/ORCLCDB as sysdba", "@healthcheck.sql"]
  #     interval: 30s
  #     timeout: 10s
  #     retries: 5

# networks:
#   cred_system_network:
#     driver: bridge

# volumes:
#   oracle-data:
#   oracle-backup: